hibike/README.md

```diff
@@ -46,75 +104,118 @@ Each Smart Device will be assigned an 88-bit UID with the following data.
     |      (16 bits)    |   (8 bits)   |          (64 bits)          |
     +-------------------+--------------+-----------------------------+
 
-Device Type - 16-bit ID specifying the type device the Smart Device controller is attached to.
-              Device types are enumerated in Section 4
+- Device Type
+    - 16-bit ID specifying the Type of a Smart Device
+    - Device types are enumerated in Section 5
+
+- Year
+  - 8-bit ID corresponding to the competition year that the Smart Device was manufactured for. 
+  - The 2015-2016 season will correspond to 0x00
+
+- ID
+    - Randomly generated 64-bit number that will uniquely identify each Smart Device within a specific device type and year.
+    - With 64-bit IDs, the probability of a hash collision with 1000 of 1 type of device per year is roughly 0.05%
+
+
+## Section 3: Parameters and bitmaps
+- Hibike abstracts every smart device as a set of parameters that map to values
+- Each smart device contains some number of paramaters, which can be read/written to.
 
-Year        - 8-bit ID corresponding to the competition year that the Smart Device was manufactured
-              for. The 2015-2016 season will correspond to 0x00
+- Paramaters can have many types, such as bool, int, short, byte, float, etc.
+- Some paramaters are read only, some are write only, and some support both.
+- A config file will describe the paramaters for each Device Type (name, type, permissions).
+- Some packets encode sets of parameters in the form of bitmaps.
 
-ID          - Randomly generated 64-bit number that will uniquely identify each Smart Device within
-              a specific device type and year. With 64-bit IDs, the probability of a hash collision 
-              with 1000 of 1 type of device per year is roughly 0.05%
 
-## Section 3: Enumerations
+
+
+    ```
+    +-------------------+-----------------------------+     +-----------------------------+
+    |      Params       |         Value 0             | ... |         Value 15            |
```

Comment:
Let's say you only request 2 params via the bitmask. Does the value section only contain data for two values, or will there be 16 value sections (with most of them set to zero)? I think this aspect of the spec can be clarified.